src/goto-symex/symex_main.cpp

```diff
@@ -54,6 +54,31 @@ void goto_symext::claim(const expr2tc &claim_expr, const std::string &msg)
   if(is_true(new_expr))
     return;
 
+  bool is_violated = false;
+
+  if(options.get_bool_option("smt-symex-assert"))
+  {
+    auto rte = std::dynamic_pointer_cast<runtime_encoded_equationt>(target);
+    equality2tc question(gen_true_expr(), new_expr);
+    try
+    {
+      // check whether the assertion holds
+      tvt res = rte->ask_solver_question(question);
+      if(res.is_true())
+        // we don't add this assertion to the resulting VCs
+        return;
+      else if(res.is_false())
+        // this assertion is violated
+        is_violated = true;
+      else
+        this->msg.status("Incremental verification returned unknown");
+    }
+    catch(runtime_encoded_equationt::dual_unsat_exception &e)
+    {
+      assert(0 && "Can't check this assertion, sorry");
```

Comment:
Who throws this exception? Does it give more info? If so, try to explain the reason why it was throw.